### PR TITLE
fix: NPE when Chart.category() is null [DHIS2-8590]

### PR DESCRIPTION
**Pull request by maikelarabori**
Backporting the NPE workaround to (2.32)

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/chart/impl/DefaultChartService.java [hunk 1]
```diff
@@ -796,7 +798,9 @@ else if ( chart.isAnalyticsType( AnalyticsType.EVENT ) )
         valueMap = DimensionalObjectUtils.getSortedKeysMap( valueMap );
 
         List<NameableObject> seriez = new ArrayList<>( chart.series() );
-        List<NameableObject> categories = new ArrayList<>( chart.category() );
+
+        List<DimensionalItemObject> dimensionalItemObjects = chart.category() != null ? chart.category() : emptyList();
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/chart/impl/DefaultChartService.java:802]: Do we know why this is null? I.e. in what circumstances is it null? I thought category should never be null.

Comment (maikelarabori) [reply dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/chart/impl/DefaultChartService.java:802]: Some Charts like Pie or Gauge have only series. This chart generation is not prepared to support some kind of charts...

Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/chart/impl/DefaultChartService.java:802]: Could we do:

`List<NameableObject> categories = new ArrayList<>( ObjectUtils.firstNonNull( chart.category(), new ArrayList<>() ) );

Comment (maikelarabori) [reply dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/chart/impl/DefaultChartService.java]: Thanks Lars. I applied the same code I applied for other branches just to keep them in sync. It's pretty much the same in the end.

**Pull request merged**